packages/fast-usdc-contract/src/fast-usdc.flows.ts

```diff
@@ -11,3 +37,97 @@ export const makeNobleAccount = (async (orch: Orchestrator) => {
   return nobleChain.makeAccount();
 }) satisfies OrchestrationFlow;
 harden(makeNobleAccount);
+
+export const forwardFunds = async (
+  orch: Orchestrator,
+  {
+    currentChainReference,
+    supportsCctp,
+    log = makeTracer('FlowForwardFunds'),
+    getNobleICA,
+    settlementAccount,
+    statusManager,
+  }: Context,
+  tx: {
+    txHash: EvmHash;
+    amount: NatAmount;
+    destination: AccountId;
+  },
+) => {
+  await null;
+  assertAllDefined({
+    currentChainReference,
+    supportsCctp,
+    log,
+    getNobleICA,
+    settlementAccount,
+    statusManager,
+    tx,
+  });
+  const { amount, destination, txHash } = tx;
+  log('trying forward for', amount, 'to', destination, 'for', txHash);
+
+  const { namespace, reference } = parseAccountId(destination);
+
+  const settlement = await settlementAccount;
+  const intermediateRecipient = getNobleICA().getAddress();
+
+  if (reference === currentChainReference) {
```

Comment:
Is there any way there could be an identical reference in a non-cosmos chain? Previously the `currentChainReference` was only checked for the cosmos namespace.

Reply:
Good thinking. It's exceeding unlikely that any other ecosystem would have `agoric-3` but it's not an assumption we have to make. I'll revise